workflow/schemas.py

```diff
@@ -296,6 +298,32 @@ def _corners_to_array(corners_spec: list[dict[str, float]]) -> np.ndarray:
         Literal("resolution", description="Subdivision resolution."): And(
             float, _is_positive
         ),
+        Literal("stype", description="Slip type for generic_slip2srf"): Or(
+            None, 
+            And(str, lambda s: s in ALLOWED_STYPE_VALUES)
```

Comment:
You should actually convert this to the correct type:

```suggestion
            Use(Stype)
```

This has the positive side-effect of meaning you don't need the dictionary to check if it is an allowed value, because `Stype(string)` will raise an exception if `string` is not the name of one of its members.